src/main/java/org/jabref/logic/importer/fetcher/ApsFetcher.java

```diff
@@ -0,0 +1,82 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.jabref.logic.importer.FulltextFetcher;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.identifier.DOI;
+
+import kong.unirest.Unirest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * FulltextFetcher implementation that attempts to find a PDF URL at APS.
+ *
+ * @see 'http://harvest.aps.org/docs/harvest-api'
+ */
+public class ApsFetcher implements FulltextFetcher {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApsFetcher.class);
+
+    // The actual API needs either an API key or a header. This is a workaround.
+    private static final String DOI_URL = "https://www.doi.org/";
+    private static final String PDF_URL = "https://journals.aps.org/prl/pdf/";
+
+    @Override
+    public Optional<URL> findFullText(BibEntry entry) throws IOException {
+        Objects.requireNonNull(entry);
+
+        Optional<DOI> doi = entry.getDOI();
```

Comment:
Rewrote it like that. A bit unsure about throwing an unchecked exception, is that fine? I guess those are supposed to be handled elsewhere, since the method interface throws them.